doc/configuration.md

```diff
@@ -0,0 +1,288 @@
+### Configuration Loading Order
+
+#### 1. CLI Override (`--config`)
+   - The `--config` flag specifies the path to load the configuration from, which can be either a single file or a directory.
+     - **File:** If the path points to a file, only that file is loaded.
+     - **Directory:** If the path points to a directory, a series of `.toml` files in the specified directory will be loaded and merged based on specific rules.
+   - **Example:** `azure-init --config /path/to/custom-config.toml`
+
+#### 2. Directory Loading Logic
+   - If the `--config` parameter points to a directory, `azure-init` follows this hierarchy:
+     - First, it looks for a base file named `azure-init.toml` in the directory.
+     - Then, it merges any additional `.toml` files in a `.d` subdirectory within the specified directory, in lexicographical order.
+     ```text
+     /etc/azure-init/
+     ├── azure-init.toml            # Base configuration
+     └── .d/
+         ├── 01-network.toml        # Additional network configuration
+         ├── 02-ssh.toml            # Additional SSH configuration
+         └── 99-overrides.toml      # Final overrides
+     ```
+
+#### 3. Defaults in Code
+   - If neither a file nor a directory is provided, `azure-init` falls back to using the default values specified in `config.rs`.
+
+### Example 1: Single Configuration File
+**Command:**
+```sh
+azure-init --config /path/to/custom-config.toml
+```
+**Order of Merging:**
+1. Loads `/path/to/custom-config.toml`.
+2. Applies any CLI overrides if present.
+3. Fills in missing values with defaults from `config.rs`.
+
+### Example 2: Directory with Multiple .toml Files
+
+**Command:**
+```sh
+azure-init --config /path/to/custom-config-directory
+```
+**Directory Structure**
+```bash
+/path/to/custom-config-directory/
+├── azure-init.toml                # Base configuration
+└── .d/
+    ├── 01-network.toml            # Network configuration
+    ├── 02-ssh.toml                # SSH configuration
+    └── 99-overrides.toml          # Overrides
+```
+**Order of Merging:**
+
+1. Loads `azure-init.toml` as the base configuration.
+2. Merges `.d` files in alphabetical order:
+   - `01-network.toml`
+   - `02-ssh.toml`
+   - `99-overrides.toml`
+3. Applies any CLI overrides if present.
+4. Fills in missing values with defaults from `config.rs`.
+
+### Example 3: Default Path without --config
+**Assumption:** If no `--config` path is specified, `azure-init` will only use the default values set within `config.rs`.
+
+**Order of Loading:**
+
+1. Loads configuration directly from defaults specified in `config.rs`.
+
+## Validation Behaviors
+
+Custom enum types for fields validate that config settings set by the user match the supported options. If a user enters an unsupported config value, 
+deserialization will fail because Serde will not be able to map that value to one of the enum variants.
+
+### Example: 
+```toml
+# Configure the query_sshd_config field.
+[ssh]
+query_sshd_config = false
```

Comment:
is this an unsupported value?  it's not clear from the example

Reply:
regardless, elaborate on what will happen on validation failure

e.g. service will fail with non-zero exit code, and failure will be reported to Azure platform